.github/workflows/test_java_cookbook.yml

```diff
@@ -54,6 +52,51 @@ jobs:
           sudo apt update
           sudo apt install libcurl4-openssl-dev libssl-dev
       - name: Run tests
-        run: make javatest
+        run: |
+          if [ "${{ github.base_ref }}" == "main" ]; then
+            export ARROW_NIGHTLY=1
+          else
+            export ARROW_NIGHTLY=0
+          fi
+          make javatest
+      - name: Build cookbook
+        run: |
+          if [ "${{ github.base_ref }}" == "main" ]; then
+            export ARROW_NIGHTLY=1
+          else
+            export ARROW_NIGHTLY=0
+          fi
```

Comment:
You could move this to to env using gha expressions:
```yaml
env:
  ARROW_R_NIGHTLY: ${{ github.base_ref == 'main' && 1 || 0 }}
```

Reply:
(same for the other occurences ofc) the general foirm is `${{ <condition> && <string_true> || <string_false> }}`